haskell/nettest/Nettest.hs

```diff
@@ -80,17 +81,14 @@ scNettestScenario constructInitialStorage stablecoinContract originateTransferli
   let
     expectFailed = flip expectFailure NettestFailedWith
 
-    sc :: AddressOrAlias
-    sc = AddressResolved scAddress
-
     tp :: Address -> Address -> Natural -> FA2.TransferParams
     tp from to value = constructSingleTransfer
       (#from_ .! from)
       (#to_ .! to)
       (#amount .! value)
 
     callTransferWithOperator op from to value =
-      callFrom (AddressResolved op) sc (ep "transfer") (tp from to value)
+      callFrom (AddressResolved op) sc DefEpName (Call_FA2 . FA2.Transfer $ tp from to value)
```

Comment:
It may be out of scope of this PR, but I like the previous version more. There `tezos-client` and RPC would be called with a small argument value and explicit entrypoint name, while in this version we always pass a large value wrapped into `Right` and `Left` constructors. So the previous version at least has cleaner logs. I am not sure whether there will be difference in block explorers, i. e. whether they can realize that `transfer` is called when you call the default entrypoint and pass `Left (Right (blabla…`. Anyway, I think the previous version should be preferred at least for cleaner logs.

AFAIU it's done this way because of [this](https://gitlab.com/morley-framework/morley/-/merge_requests/499#note_388408872). So probably the problem is out of scope of this PR and what we have now is the only possible fast solution.

Reply:
Agreed, the previous version looks a lot better. I'll make a note to come back to this after addressing the comments in morley's !499.

Reply:
FYI, I ended up reverting these changes. Apparently, calling `DefEpName` with `Call_FA2 (FA2.Transfer x)` is not equivalent to calling `ep "transfer"` with `x`. I was under the impression that they were equivalent, but it turns out I was getting [errors like this](https://buildkite.com/serokell/stablecoin/builds/737#e9d58043-8051-463d-a43f-65461f2a93b0/63-109):

```
src/Morley/Nettest/Pure.hs:52
  | Interpreter failed: The contract parameter is ill-typed: Error checking value Left { Pair "tz1f1S7V2hZJ3mhj47djb5j1saek8c2yB2Cx" { Pair "tz1f1S7V2hZJ3mhj47djb5j1saek8c2yB2Cx" (Pair 0 111) } } against type unit: unknown value Error occured on line 1 char 1.
```

I added `callFrom'` [here](https://github.com/tqtezos/stablecoin/pull/79/files#diff-80418a2f9a5a7631759638be01c0d856R348-R366) as a temporary measure. I'll come back to this later with a more permanent solution.